prov/verbs/src/ep_rdm/verbs_av_ep_rdm.c

```diff
@@ -190,10 +205,11 @@ static int fi_ibv_rdm_av_remove(struct fid_av *av, fi_addr_t * fi_addr,
 			conn, inet_ntoa(conn->addr.sin_addr),
 			ntohs(conn->addr.sin_port));
 		HASH_DEL(fi_ibv_rdm_tagged_conn_hash, conn);
-		fi_ibv_rdm_start_disconnection(conn);
+		err = fi_ibv_rdm_start_disconnection(conn);
+		ret = (ret == FI_SUCCESS) ? err : ret;
```

Comment:
ret is guaranteed to be FI_SUCCESS only at first iteration. We should initiate disconnection for all requested connections to prevent leaks but only first occurred unsuccessful error code will be returned.